ziti-springboot-client/src/main/java/org/openziti/springboot/client/web/config/ZitiHttpClientConfiguration.java

```diff
@@ -194,16 +192,34 @@ public void destroy() {
     Optional.ofNullable(zitiSSLConnectionSocketFactory).ifPresent(ZitiSSLConnectionSocketFactory::shutdown);
   }
 
-  private static DnsResolver getDnsResolver() {
+  @ConditionalOnProperty(value = "spring.ziti.client.dns-resolver.enabled", havingValue = "true", matchIfMissing = true)
+  @Bean
+  public static DnsResolver zitiDnsResolver() {
     return new DnsResolver() {
       @Override
-      public InetAddress[] resolve(String s) throws UnknownHostException {
-        return new InetAddress[]{InetAddress.getLocalHost()};
+      public InetAddress[] resolve(String hostname) throws UnknownHostException {
+        log.debug("Resolving {} using Ziti DNS Resolver", hostname);
+        InetAddress address = Ziti.getDNSResolver().resolve(hostname);
+        if (address != null) {
+          log.debug("Resolved {} using Ziti to address {}", hostname, address);
+          return new InetAddress[]{address};
+        }
+
+        log.debug("Unable to resolve {} using Ziti DNS Resolver, trying regular DNS resolution", hostname);
+        return new InetAddress[]{InetAddress.getByName(hostname)};
       }
 
       @Override
-      public String resolveCanonicalHostname(String s) throws UnknownHostException {
-        return InetAddress.getLocalHost().getCanonicalHostName();
+      public String resolveCanonicalHostname(String hostname) throws UnknownHostException {
+        log.debug("Resolving {} using Ziti DNS Resolver", hostname);
```

Comment:
all this logging will happen anyway, won't it all be duplicated when .resolve is called?

Reply:
All the logging i saw is `trace` level but it also doesn't log that it is falling back to os resolution which i think is useful for applications.